src/cli/domain/init-template/install-template.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const Spinner = require('cli-spinner').Spinner;
+const spawn = require('cross-spawn');
+const path = require('path');
+const _ = require('lodash');
+const process = require('process');
+const strings = require('../../../resources');
+
+module.exports = function installTemplate(options, scaffold) {
+  const {
+    componentName,
+    templateType,
+    compiler,
+    cli,
+    componentPath,
+    local,
+    template,
+    logger,
+    callback
+  } = options;
+
+  const installing = new Spinner(
+    strings.messages.cli.installCompiler(compiler, local)
+  );
+  installing.start();
+
+  const args = {
+    npm: [
+      'install',
+      '--save-dev',
+      '--save-exact',
+      local ? path.resolve(process.cwd(), templateType) : compiler
+    ]
+  };
+
+  const installProc = spawn(cli, args[cli], { cwd: componentPath });
+
+  installProc.on('error', () => callback('template type not valid'));
```

Comment:
should we `installing.stop(true)` in case of error?
Also, can we move this messages to `resources`? I think usually the previous convention was to return either a message from `resources` or a code like `TEMPLATE_NOT_VALID` so that later you print the `resources.errors[error_code]....` in a lower level like the facade.